sw/airborne/modules/optical_flow/px4flow.c

```diff
@@ -30,64 +31,134 @@
 #include "modules/datalink/mavlink_decoder.h"
 #include "subsystems/abi.h"
 
+// Messages
+#include "mcu_periph/uart.h"
+#include "pprzlink/messages.h"
+#include "subsystems/datalink/downlink.h"
+
+struct mavlink_heartbeat heartbeat;
 struct mavlink_optical_flow optical_flow;
-bool optical_flow_available;
+struct mavlink_optical_flow_rad optical_flow_rad;
 
 // message ID in Mavlink (v1.0)
+#define MAVLINK_OPTICAL_HEARTBEAT_MSG_ID 0
 #define MAVLINK_OPTICAL_FLOW_MSG_ID 100
-// size of the structure
-#define MAVLINK_OPTICAL_FLOW_LEN 26
+#define MAVLINK_OPTICAL_FLOW_RAD_MSG_ID 106
 
-#ifndef PX4FLOW_NOISE
-#define PX4FLOW_NOISE 0.5
+/*
+ * acceptable quality of optical flow (0-min,1-max)
+ * higher threshold means fewer measurements will be available
+ * but they will be more precise. Generally it is probably a better
+ * idea to have lower threshold, and propagate the noise (1-quality)
+ * to the INS
+ */
+#ifndef PX4FLOW_QUALITY_THRESHOLD
+#define PX4FLOW_QUALITY_THRESHOLD 0.1
```

Comment:
Well do you have a better solution? "scale somehow" is essentially what I am already doing here with the user defined scaling factor.